sklearn/utils/estimator_checks.py

```diff
@@ -426,7 +428,15 @@ def check_estimator(Estimator, generate_only=False):
     if generate_only:
         return checks_generator
 
+    xfail_checks = _safe_tags(estimator, '_xfail_test')
+
     for estimator, check in checks_generator:
+        check_name = _set_check_estimator_ids(check)
+        if xfail_checks and check_name in xfail_checks:
+            # skip tests marked as a known failure and raise a warning
+            msg = xfail_checks[check_name]
+            warnings.warn(f'Skipping {check_name}: {msg}', SkipTestWarning)
+            continue
```

Comment:
This is a change following https://github.com/scikit-learn/scikit-learn/pull/16502 analogous to what was added in `test_common.py::test_estimators` . Without this `test_check_estimator_clones` started to fail with this PR, since it runs `check_estimator` on `MiniBatchKMeans` which now has one xfail test.

Skipping with a warning tests marked as xfail in `check_estimator`, as done here, is a solution to this issue.